src/CMakeLists.txt

```diff
@@ -103,10 +103,36 @@ endif()
 
 
 option(ENABLE_GLES "Use OpenGL ES instead of OpenGL" FALSE)
-mark_as_advanced(ENABLE_GLES)
+set(GLES_VERSION "2" CACHE STRING "Specify OpenGL ES Version (1, 2, or BOTH)")
+mark_as_advanced(ENABLE_GLES GLES_VERSION)
+
+option(ENABLE_TOUCH "Enable Touchscreen support" FALSE)
+if(ENABLE_TOUCH)
+	if(NOT ENABLE_GLES)
+		message(FATAL_ERROR "OpenGL ES Support Is Required For Touchscreen Support")
+	endif()
+	add_definitions(-DHAVE_TOUCHSCREENGUI)
+endif()
+
 if(BUILD_CLIENT)
+	if(ENABLE_TOUCH)
+		find_library(XINPUT_LIBRARY Xi REQUIRED)
+	endif()
 	if(ENABLE_GLES)
-		find_package(OpenGLES2 REQUIRED)
+		find_package(EGL REQUIRED)
+
+		set(VALID_GLES FALSE)
+		if("${GLES_VERSION}" STREQUAL "2" || "${GLES_VERSION}" STREQUAL "BOTH")
+			find_package(OpenGLES2 REQUIRED)
+			set(VALID_GLES TRUE)
+		endif()
+		if("${GLES_VERSION}" STREQUAL "1" || "${GLES_VERSION}" STREQUAL "BOTH")
+			find_package(OpenGLES1 REQUIRED)
+			set(VALID_GLES TRUE)
+		endif()
+		if(NOT VALID_GLES)
+			message(FATAL_ERROR "Invalid OpenGL ES Version: ${GLES_VERSION}")
+		endif()
```

Comment:
Nitpicking, but I think the previous `if / elseif / endif` way of writing this logic was better.

Reply:
I spun this out into its own PR: #11424.